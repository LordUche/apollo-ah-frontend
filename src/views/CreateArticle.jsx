import React from 'react';
import ReactQuill from 'react-quill';
import TagsInput from './TagsInput';

/**
 * Create Article view component
 * @param {object} props Component props
 * @param {function} props.updateInputHandler updating input values
 * @param {function} props.updateBodyHandler updating the state of the article body
 * @param {function} props.imageUploadHandler submit image to cloudinary
 * @param {function} props.submitHandler submit event handler
 * @param String props.valueText updated form data body
 * @param Boolean props.isLoading updated loading state passed as props
 * @returns {import('@babel/types').JSXAttribute}
 */

function CreateArticle(props) {
  const {
    updateInputHandler,
    updateBodyHandler,
    imageUploadHandler,
    submitHandler,
    valueText,
    isLoading,
  } = props;

  let { responseMessage } = props;

  if (responseMessage === 'title is not a string') {
    responseMessage = 'title is required';
  }

  const handleUpdate = (updatedTags) => {
    props.updateTagHandler(updatedTags);
  };

  const tagsUpdateHandler = (updatedTags) => {
    handleUpdate(updatedTags);
  };

  return (
    <div>
      <header className="page-header" data-pg-collapsed>
        <h3>New Article</h3>
      </header>
      <div className="card mb-5">
        <form className="form p-sm-4 p-3 form-borderless">
          <div className="form-group mb-4" data-pg-collapsed>
            <label htmlFor="title" className="font-weight-bold">Title</label>
            <input type="text" className="form-control text-capitalize" id="title" name="title" onChange={updateInputHandler} required />
          </div>
          <div className="form-group mb-4" data-pg-collapsed>
            <label htmlFor="description" className="font-weight-bold">Short Description</label>
            <textarea className="form-control textarea-sm" id="description" name="description" onChange={updateInputHandler} required />
          </div>
          <div className="form-group mb-4" data-pg-collapsed>
            <label htmlFor="description" className="font-weight-bold">Details</label>
            <ReactQuill value={valueText} onChange={updateBodyHandler} />
          </div>
          <div className="form-group mb-4" data-pg-collapsed>
            <label className="font-weight-bold" htmlFor="category">Category</label>
            <select onChange={updateInputHandler} defaultValue="DEFAULT" className="custom-select" name="categoryId" id="category">
              <option value="DEFAULT">-- Choose one --</option>
              <option value="3">Business</option>
              <option value="7">Entertainment</option>
              <option value="4">Fashion</option>
              <option value="6">Food</option>
              <option value="5">Health</option>
              <option value="8">History</option>
              <option value="1">Technology</option>
              <option value="2">Sports</option>
              <option value="9">Others</option>
            </select>
          </div>
          <div className="form-group mb-4" data-pg-collapsed>
            <label className="font-weight-bold" htmlFor="tags">Cover Image</label>
            <button type="button" onClick={imageUploadHandler} className="create-article-image-btn">
              <div className="camera-icon" />
            </button>
          </div>
          <div className="form-group mb-4" data-pg-collapsed>
            <label className="font-weight-bold" htmlFor="tags">Tags (optional)</label>
            <TagsInput tagsUpdateHandler={tagsUpdateHandler} />
          </div>
          <button
            type="submit"
            onClick={submitHandler}
            style={{ backgroundColor: '#66008c', color: 'white' }}
            className="btn btn-brand btn-plain mt-3 create-article-btn"
          >
            {
              isLoading ? <div className="spinner-border text-light" /> : 'Publish article'
            }
          </button>
          {
            responseMessage && <div className="alert alert-danger create-article-alert">{responseMessage}</div>
          }
        </form>
      </div>
    </div>
  );
}


export default CreateArticle;
