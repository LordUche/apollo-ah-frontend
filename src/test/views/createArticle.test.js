import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore, mockState } from '../setup';
import CreateArticle from '../../views/CreateArticle';

createMockStore();
const { profile } = mockState.user;


describe('<CreateArticle>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(
      <CreateArticle
        isLoading={false}
        handleUpdateProfile={jest.fn()}
        profile={profile}
        token="encoded"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with errorData', () => {
    mockState.user.errorData = [{}];
    const wrapper = shallow(
      <CreateArticle
        updateInputHandler={jest.fn}
        updateBodyHandler={jest.fn()}
        updateTagHandler={jest.fn()}
        imageUploadHandler={jest.fn()}
        submitHandler={jest.fn()}
        valueText="article body"
        isLoading={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
