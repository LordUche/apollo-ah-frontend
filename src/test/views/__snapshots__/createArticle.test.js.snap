// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateArticle> should render with errorData 1`] = `ShallowWrapper {}`;

exports[`<CreateArticle> should render without crashing 1`] = `ShallowWrapper {}`;
